CMakeLists.txt

```diff
@@ -0,0 +1,150 @@
+  # can be troublesome. Note that nanobind requires FindPython.
+  set(Python_EXECUTABLE "${Python3_EXECUTABLE}")
+  find_package(Python 3.9 COMPONENTS Interpreter Development.Module REQUIRED)
+endif()
+
+if(NOT "${Python_EXECUTABLE}" STREQUAL "${Python3_EXECUTABLE}")
+  message(WARNING "FindPython and FindPython3 found different executables. You may need to pin -DPython_EXECUTABLE and -DPython3_EXECUTABLE (${Python_EXECUTABLE} vs ${Python3_EXECUTABLE})")
+endif()
+
+#-------------------------------------------------------------------------------
+# MLIR/LLVM Dependency
+#-------------------------------------------------------------------------------
+
+# Both the Lighthouse and MLIR Python bindings require nanobind. We initialize
+# it here at the top level so that everything uses ours consistently.
+if(LIGHTHOUSE_BUILD_PYTHON_BINDINGS)
+  # TODO: Configure a flag to use system deps directly.
+  include(FetchContent)
+  FetchContent_Declare(
+      nanobind
+      GIT_REPOSITORY https://github.com/wjakob/nanobind.git
+      GIT_TAG        0f9ce749b257fdfe701edb3cf6f7027ba029434a # v2.4.0
+  )
+  FetchContent_MakeAvailable(nanobind)
+endif()
+
+# Get the main LLVM deps.
+# TODO: Add a flag to use installed llvm instead.
+# TODO: There are a lot of default flags that are missing here. The build
+# system was only configured enough to build mlir python bindings.
+
+# Enable MLIR Python bindings if requested.
+set(MLIR_ENABLE_BINDINGS_PYTHON OFF CACHE BOOL "")
+if (LIGHTHOUSE_BUILD_PYTHON_BINDINGS)
+  set(MLIR_ENABLE_BINDINGS_PYTHON ON)
+endif()
+# Disable MLIR attempting to configure Python dev packages. We take care of
+# that in Lighthouse as a super-project.
+set(MLIR_DISABLE_CONFIGURE_PYTHON_DEV_PACKAGES ON CACHE BOOL "" FORCE)
+
+# LLVM defaults to building all targets. We always enable targets that we need
+# as we need them, so default to none. The user can override this as needed,
+# which is fine.
+set(LLVM_TARGETS_TO_BUILD "" CACHE STRING "")
+
+# These defaults are moderately important to us, but the user *can*
+# override them (enabling some of these brings in deps that will conflict,
+# so ymmv).
+set(LLVM_INCLUDE_EXAMPLES OFF CACHE BOOL "")
+set(LLVM_INCLUDE_TESTS OFF CACHE BOOL "")
+set(LLVM_INCLUDE_BENCHMARKS OFF CACHE BOOL "")
+set(LLVM_APPEND_VC_REV OFF CACHE BOOL "")
+set(LLVM_ENABLE_IDE ON CACHE BOOL "")
+set(LLVM_ENABLE_BINDINGS OFF CACHE BOOL "")
+
+# Force LLVM to avoid dependencies, which we don't ever really want in our
+# limited builds.
+set(LLVM_ENABLE_LIBEDIT OFF CACHE BOOL "Default disable")
+set(LLVM_ENABLE_LIBXML2 OFF CACHE BOOL "Default disable")
+set(LLVM_ENABLE_TERMINFO OFF CACHE BOOL "Default disable")
+set(LLVM_ENABLE_ZLIB OFF CACHE BOOL "Default disable")
+set(LLVM_ENABLE_ZSTD OFF CACHE BOOL "Default disable")
+set(LLVM_FORCE_ENABLE_STATS ON CACHE BOOL "Default enable")
+
+# Unconditionally enable mlir.
+set(LLVM_ENABLE_PROJECTS mlir)
```

Comment:
If the user sets more project on the command line, doesn't this "reset" back to just MLIR?